en/guide/toolchain.md

```diff
@@ -0,0 +1,121 @@
+
+Below we explain how to set up a minimal make file for your application.
+
+> **Tip** You can also kick off your application by copying modifying our existing [example code](https://github.com/dronecore/DroneCore/blob/master/example/takeoff_land/). Note that if you move your new application outside of the example directory you will need to update the (relative) paths to the locally installed DroneCore library.
+
+
+## Build Definition File - CMakeLists.txt
+
+*Cmake* uses a definition file named **CMakeLists.txt** to define the project. This specifies the name of the project, compiler flags for different platforms and targets, where to find dependencies (libraries and header files), source files to build, and the name of the generated binary. **CMakeLists.txt** is typically stored in the root directory of your app project.
+
+The first two lines of the file should specify the minimum compatible version of *cmake* and *your* project's name. 
+```cmake
+cmake_minimum_required(VERSION 2.8.12)
+project(your_project_name)
+```
+
+Next add the define flags that are used in compilation on different platforms 
+(you can treat this as boilerplate - it enables relatively strict handling of warnings).
+```cmake
+if(NOT MSVC)
+    add_definitions("-std=c++11 -Wall -Wextra -Werror")
+else()
+    add_definitions("-std=c++11 -WX -W2")
+    set(platform_libs "Ws2_32.lib")
+endif()
+
+# Add DEBUG define for Debug target
+set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG")
+```
+
+Add the line below to find a thread library on Linux, Mac, and Windows. This is required in order to use DroneCore!
+```cmake
+find_package(Threads REQUIRED)
+```
+
+The `include_directories()` command tells *cmake* where to find locally installed header files (if DroneCore is installed globally this is not required). 
+
+```cmake
+include_directories(
+    # Not needed if installed system-wide
+    ${CMAKE_SOURCE_DIR}/../../install/include
+)
+```
+
+> **Note** The fragment above is taken from one of the DroneCore examples, and specifies a relative path from an example sub-directory to where DroneCore exports the header files by default. 
+> If you have installed DroneCore locally and your application is not located alongside the DroneCore examples you will need to change this. 
+>
+>  Note that `CMAKE_SOURCE_DIR` is the directory in which **CMakeLists.txt** is stored, and provides a useful way of defining relative paths to the header files.
+
+Then use `add_executable()` to specify the name for your generated executable and your source files. 
+```cmake
+add_executable(your_executable_name
+    your_source_file.cpp
+)
+```
+
+The final sections of the file defines the libraries to link against. 
+The first part is used if DroneCore is installed locally, and will have to be modified depending on the relative path from your code to the **DroneCore/install/** directory.
```

Comment:
@julianoes Section below here is my "guess" on what the file does. Can you please check this a bit more carefully.

Reply:
I think you're right. Just the paths are strange. The reason is that we install it locally inside `install`, instead we should probably support the usual install into `/usr/local/`, at least for Linux/Mac.

I made an issue for this: https://github.com/dronecore/DroneCore/issues/70